### ISSUE TITLE
about training loss

### ISSUE THREAD

**Issue by johnlanbor** [eriklindernoren/PyTorch-YOLOv3#18]
when i train with coco or my own data
the loss is not convergence especially for w and h

371: nGT 19, recall 13, AP 68.42% proposals 0, loss: x 3.118402, y 3.105752, w nan, h nan, conf 164.831421, cls 83.258507, total nan
372: nGT 42, recall 11, AP 26.19% proposals 0, loss: x 3.763822, y 4.745901, w nan, h nan, conf 174.948318, cls 122.696747, total nan
372: nGT 42, recall 21, AP 50.00% proposals 0, loss: x 6.076531, y 5.691019, w nan, h nan, conf 317.763611, cls 157.752930, total nan
372: nGT 42, recall 34, AP 80.95% proposals 0, loss: x 6.518974, y 5.134117, w nan, h nan, conf 478.700623, cls 175.281006, total nan
373: nGT 7, recall 5, AP 71.43% proposals 0, loss: x 0.831325, y 1.967510, w nan, h nan, conf 55.965015, cls 30.674187, total nan
373: nGT 7, recall 7, AP 100.00% proposals 0, loss: x 1.058911, y 1.067211, w nan, h nan, conf 101.454453, cls 30.674187, total nan
373: nGT 7, recall 7, AP 100.00% proposals 0, loss: x 0.877475, y 0.707009, w nan, h nan, conf 135.526093, cls 30.674187, total nan
374: nGT 29, recall 7, AP 24.14% proposals 0, loss: x 4.089700, y 4.640538, w nan, h nan, conf 129.530502, cls 113.932693, total nan
374: nGT 29, recall 12, AP 41.38% proposals 0, loss: x 3.668809, y 6.106874, w nan, h nan, conf 198.104630, cls 127.078773, total nan
374: nGT 29, recall 21, AP 72.41% proposals 0, loss: x 4.712764, y 5.478906, w nan, h nan, conf 295.735779, cls 127.078773, total nan
375: nGT 15, recall 4, AP 26.67% proposals 0, loss: x 2.289522, y 1.391438, w nan, h nan, conf 64.979248, cls 48.202293, total nan
375: nGT 15, recall 7, AP 46.67% proposals 0, loss: x 2.598335, y 2.003327, w nan, h nan, conf 118.649559, cls 61.348373, total nan
375: nGT 15, recall 11, AP 73.33% proposals 0, loss: x 3.218783, y 3.078263, w nan, h nan, conf 178.149246, cls 65.730400, total nan
376: nGT 14, recall 2, AP 14.29% proposals 0, loss: x 1.407205, y 2.768980, w nan, h nan, conf 46.998051, cls 48.202293, total nan
376: nGT 14, recall 6, AP 42.86% proposals 0, loss: x 1.456108, y 3.255125, w nan, h nan, conf 91.687042, cls 56.966347, total nan
376: nGT 14, recall 8, AP 57.14% proposals 0, loss: x 2.827966, y 2.196316, w nan, h nan, conf 131.248718, cls 61.348373, total nan
377: nGT 10, recall 7, AP 70.00% proposals 0, loss: x 0.662892, y 1.921333, w nan, h nan, conf 107.326248, cls 43.820267, total nan
377: nGT 10, recall 9, AP 90.00% proposals 0, loss: x 0.840957, y 1.428768, w nan, h nan, conf 140.404282, cls 43.820267, total nan
377: nGT 10, recall 10, AP 100.00% proposals 0, loss: x 0.755548, y 1.235753, w nan, h nan, conf 174.077911, cls 43.820267, total nan

**Comment by johnlanbor** [comment]
I think that may because the base lr is too large
when i change to 0.0001, the loss is correct

**Comment by alexandru-dinu** [comment]
Are you computing loss for w/h directly or sqrt(w)/sqrt(h)?

**Comment by johnlanbor** [comment]
@alexandru-dinu  I tried both, all got nan

**Comment by alexandru-dinu** [comment]
In the paper, they use sqrt for width and height loss calculation, but in order to do that w/h values must be positive. Sqrt of a negative number will give you nan. 

Did you apply any transformation on the w/h values predicted by the network? That is, did you compare transformed (`bx, by, bw, bh`) or raw (`tx, ty, tw, th`) [values](https://i.imgur.com/6TEg6Z1.png) with the ground truths?

**Comment by eriklindernoren** [comment]
This should be solved in the latest commit. Model does not converge unfortunately but training is stable and shouldn't result in nan:s for batch size 8.
